janusgraph-core/src/main/java/org/janusgraph/diskstorage/BackendTransaction.java

```diff
@@ -373,9 +377,10 @@ public KeyIterator edgeStoreKeys(final SliceQuery sliceQuery) {
         return executeRead(new Callable<KeyIterator>() {
             @Override
             public KeyIterator call() throws Exception {
-                return (storeFeatures.isKeyOrdered())
-                        ? edgeStore.getKeys(new KeyRangeQuery(EDGESTORE_MIN_KEY, EDGESTORE_MAX_KEY, sliceQuery), storeTx)
-                        : edgeStore.getKeys(sliceQuery, storeTx);
+                return (storeFeatures.isKeyOrdered() && !allowCustomVertexIdType)
+                    // if string vertex id is allowed, then min key and max key are unknown
```

Comment:
(nitpick)
`if string vertex id is allowed` -> `if non long-type vertex id is allowed`